pal/src/sync/cclock.cpp

```diff
@@ -12,11 +12,31 @@
 #include <errno.h>
 // do not include PAL
 
+template <typename T>
+static T AlignSize(T size, T alignment)
+{
+    return ((size + (alignment-1)) & ~(alignment-1));
+}
+template <typename T>
+static T* AlignPointer(T* ptr, size_t align = sizeof(size_t))
+{
+    return reinterpret_cast<T*>(AlignSize(reinterpret_cast<size_t>(ptr), align));
+}
+
+CCLock::CCLock(bool shouldTrackThreadId)
+{
+    char *alignedMutexPtr = AlignPointer(this->mutexPtr);
```

Comment:
Why this is needed?

Reply:
If the CCLock lands on unaligned memory, it causes SIGBUS on the next mutex manipulation on Android. I see that the OSX tests failed though...maybe I'll guard the alignment for Android only after all tests are finished.